js/fetch.ts

```diff
@@ -81,11 +88,23 @@ export class DenoHeaders implements domTypes.Headers {
     return this.headerMap.has(newname);
   }
 
+  keys(): IterableIterator<string> {
+    const iterators = this.headerMap.keys();
+    const list = Array.from(iterators);
+    return new CreateIterableIterator(list); 
```

Comment:
Is this the same ?
`return this.headerMap.keys(); `

Reply:
If we return Map's Iterator,  `Object.prototype.toString.call(iterator)` will output  `"[object Map Iterator]"`.  It is not compliant with the spec.
In browser:
``````
var headers = new Headers()
undefined
var it = headers.keys()
undefined
Object.prototype.toString.call(it)
"[object Iterator]"
Symbol.iterator in it
true
``````
So I return a new Iterator.

Reply:
``````
var m = new Map()
undefined
var mit = m.keys()
undefined
Object.prototype.toString.call(mit)
"[object Map Iterator]"
``````

Reply:
I see, even the toString conversion needs to be the same. In that case it could be nice to add a comment explaining this

Reply:
otherwhise it is confusing why iterator is cast to array then back to iterator